app/api/__init__.py

```diff
@@ -397,7 +398,7 @@
 api.route(OrdersListPost, 'order_list_post', '/orders')
 api.route(OrdersList, 'orders_list', '/orders', '/events/<int:event_id>/orders',
           '/events/<event_identifier>/orders')
-api.route(OrderDetail, 'order_detail', '/orders/<identifier>')
+api.route(OrderDetail, 'order_detail', '/orders/<identifier>', '/orders/<int:id>')
```

Comment:
Orders should not be accessible by id. The id is auto incremented and it's unsafe to use it in critical data like orders. Try mentioning identifier in self_view_kwargs when defining the schema if you are having any errors.